vulture_os/toolkit/api_parser/cisco_umbrella_managed_org/cisco_umbrella_managed_org.py

```diff
@@ -0,0 +1,269 @@
+            raise CiscoUmbrellaManagedOrgAPIError(f"Error at Cisco-Umbrella API Call URL: {self.TOKEN_URL} Code: {response.status_code} Content: {response.content}")
+        token = response.json()
+        self.cisco_umbrella_managed_org_parent_access_token = token["access_token"]
+        self.cisco_umbrella_managed_org_parent_expires_at = timezone.now() + timedelta(seconds=int(token["expires_in"]))
+
+    def _connect(self, access_token):
+        self.session = requests.Session()
+        headers = self.HEADERS
+        headers.update({'Authorization': f"Bearer {access_token}"})
+        self.session.headers.update(headers)
+
+    def _get_customers_id(self, timeout=10):
+        if not self.cisco_umbrella_managed_org_parent_access_token or not self.cisco_umbrella_managed_org_parent_expires_at or (self.cisco_umbrella_managed_org_parent_expires_at - timedelta(seconds=10) < timezone.now()):
+            self._get_parent_token()
+        self._connect(self.cisco_umbrella_managed_org_parent_access_token)
+        logger.info(f"[{__parser__}]: Getting the list of organisation id", extra={'frontend': str(self.frontend)})
+
+        response = self.session.get(self.ORGANIZATIONS_URL,
+            timeout=timeout,
+            proxies=self.proxies,
+            verify=self.api_parser_custom_certificate or self.api_parser_verify_ssl
+        )
+        if response.status_code != 200:
+            raise CiscoUmbrellaManagedOrgAPIError(f"Error at Cisco-Umbrella API Call URL: {self.ORGANIZATIONS_URL} Code: {response.status_code} Content: {response.content}")
+        return [str(customer.get("customerId")) for customer in response.json()]
+
+    def _get_customer_token(self, customer_id):
+        logger.info(f"[{__parser__}]: Getting a new authentication token for customer {customer_id}", extra={'frontend': str(self.frontend)})
+        headers = {
+            'X-Umbrella-OrgID': customer_id,
+            'Content-Type': 'application/x-www-form-urlencoded'
+        }
+        auth = f"{self.cisco_umbrella_managed_org_api_key}:{self.cisco_umbrella_managed_org_secret_key}"
+        auth_b64 = base64.b64encode(bytes(auth, encoding="utf-8")).decode()
+        headers.update({"Authorization": f"Basic {auth_b64}"})
+        response = requests.get(
+            url=self.TOKEN_URL,
+            headers=headers,
+            proxies=self.proxies
+        )
+        if response.status_code != 200:
+            raise CiscoUmbrellaManagedOrgAPIError(f"Error at Cisco-Umbrella API Call URL: {self.ORGANIZATIONS_URL} Code: {response.status_code} Content: {response.content}")
+        token = response.json()
+        self.cisco_umbrella_managed_org_customers_tokens[customer_id] = {}
+        self.cisco_umbrella_managed_org_customers_tokens[customer_id]["access_token"] = token["access_token"]
+        self.cisco_umbrella_managed_org_customers_tokens[customer_id]["expires_at"] = timezone.now() + timedelta(seconds=int(token["expires_in"]))
```

Comment:
This information is never saved back in the frontend!
1. Make sure data is saved as long as it's correctly received
2. Make sure the data is only saved when running, and not during the GUI testing phase (in that case self.frontend is None)